src/nodes/chunk_dispatch/chunk_dispatch.c

```diff
@@ -145,7 +146,14 @@ ts_chunk_dispatch_get_chunk_insert_state(ChunkDispatch *dispatch, Point *point,
 			 * postgres can do proper constraint checking.
 			 */
 			if (ts_cm_functions->decompress_batches_for_insert)
+			{
 				ts_cm_functions->decompress_batches_for_insert(cis, chunk, slot);
+				OnConflictAction onconflict_action =
+					chunk_dispatch_get_on_conflict_action(dispatch);
+				/* mark rows visible */
+				if (onconflict_action == ONCONFLICT_UPDATE)
+					dispatch->estate->es_output_cid = GetCurrentCommandId(false);
```

Comment:
I wonder - how did this worked before for the normal `insert` path? why didn't it needed `es_output_id`  before? - or for normal inserts this is not "needed" ?

I have feel that its a bit too late here to set an estate field - I think this field should be set way-way-back..but I'm not sure where to place it...

Reply:
Frankly i don't know either, i face an error as "ERROR:  attempted to update invisible tuple"
To overcome this error i had to set es_output_cid to latest current command ID.

Reply:
it should fix that for sure - and with `true` passed I think its okayish

I'll try to keep this in mind - and let you know if I've found a better place to set this...

Reply:
Looking at `ExecOnConflictUpdate()` , for the case TM_Invisible, where the error comes from, I see that 
```
test = table_tuple_lock(relation, conflictTid,
							estate->es_snapshot,
							existing, estate->es_output_cid,
							lockmode, LockWaitBlock, 0,
							&tmfd);
```
uses `estate->es_snapshot` and `estate->es_output_cid`, so I also think it makes sense to do this here.

Reply:
Thanks.